src/main/java/entities/decks/EssenceDeck.java

```diff
@@ -3,6 +3,7 @@
 import entities.cards.Card;
 import entities.cards.CardFactory;
 
+
```

Comment:
```suggestion
/**
 * EssenceDeck contains unlimited number of EssenceCards for the players to draw.
 * Thus it takes the CardFactory so that drawing an Essence is handled by creating one.
 */
```